src/test/java/menagerie/model/search/SearchRuleParserTest.java

```diff
@@ -0,0 +1,74 @@
+package menagerie.model.search;
+
+import menagerie.model.search.rules.DateAddedRule;
+import menagerie.model.search.rules.IDRule;
+import menagerie.model.search.rules.SearchRule;
+import menagerie.model.search.rules.TitleRule;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class SearchRuleParserTest {
+
+  @Test
+  void testParseIDRule() {
```

Comment:
you could split up this test case into three separate ones

Reply:
Implemented via parameterized tests now